helm/customer-managed-agent/templates/raw_deployment.yaml

```diff
@@ -0,0 +1,131 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "helm.fullname" . }}-agent-config
+  labels:
+  {{- include "helm.labels" . | nindent 4 }}
+data:
+  PULUMI_AGENT_IMAGE: {{ .Values.agentConfig.pulumiAgentImage | quote }}
+  PULUMI_AGENT_IMAGE_PULL_POLICY: {{ .Values.agentConfig.pulumiAgentImagePullPolicy
+    | quote }}
+  PULUMI_AGENT_SERVICE_URL: {{ .Values.agentConfig.pulumiAgentServiceUrl | quote
+    }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "helm.fullname" . }}-agent-secret
+  labels:
+  {{- include "helm.labels" . | nindent 4 }}
+data:
+  PULUMI_AGENT_TOKEN: {{ required "agentSecret.pulumiAgentToken is required" .Values.agentSecret.pulumiAgentToken
+    | b64enc | quote }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "helm.fullname" . }}-deployment-agent
+  labels:
+  {{- include "helm.labels" . | nindent 4 }}
+  annotations:
+    {{- toYaml .Values.deploymentAgent.serviceAccount.annotations | nindent 4 }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "helm.fullname" . }}-deployment-agent
+  labels:
+  {{- include "helm.labels" . | nindent 4 }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  - pods/log
+  - configmaps
+  verbs:
+  - create
+  - get
+  - list
+  - watch
+  - update
+  - delete
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ include "helm.fullname" . }}-deployment-agent
+  labels:
+  {{- include "helm.labels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: '{{ include "helm.fullname" . }}-deployment-agent'
+subjects:
+- kind: ServiceAccount
+  name: '{{ include "helm.fullname" . }}-deployment-agent'
+  namespace: '{{ .Release.Namespace }}'
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "helm.fullname" . }}-pool
+  labels:
+  {{- include "helm.labels" . | nindent 4 }}
+  annotations:
+    app.kubernetes.io/name: pulumi-deployment-agent-pool
+spec:
+  replicas: {{ .Values.pool.replicas }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: customer-managed-deployment-agent
+    {{- include "helm.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: customer-managed-deployment-agent
+      {{- include "helm.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
```

Comment:
There's a couple of missing knobs we're not exposing.

IMO users need to be able to set any of the underlying configuration options - but especially important ones are:
- topologySpreadConstraints,
- nodeSelector
- affinity
- side car containers
- init containers
- tolerations
- securityContext
- hostNetwork
- dnsPolicy
- imagePullSecrets
- priorityClassName
- runtimeClassName

I might have missed some - just recounted those from memory. I'd recommend checking what some of the other common helm charts do (e.g. datadog)

Reply:
Although I've not used this list as an exhaustive checklist, I've added the ones we use in the PKO as a guide.